core/io/file_access.cpp

```diff
@@ -852,6 +858,70 @@ String FileAccess::get_sha256(const String &p_file) {
 	return String::hex_encode_buffer(hash, 32);
 }
 
+void FileAccess::add_resource_path(const String &p_protocol, const String &p_path) {
+	ERR_FAIL_COND_MSG(p_protocol.length() == 0, "Protocol parameter is empty.");
+
+	String protocol = p_protocol;
+	if (protocol.ends_with("://")) {
+		protocol = protocol.left(protocol.length() - 3);
+	}
+
+	ERR_FAIL_COND_MSG(protocol == "res" || protocol == "user", vformat("Protocol \"%s://\" is built-in.", protocol));
```

Comment:
```suggestion
	ERR_FAIL_COND_MSG(protocol == "res" || protocol == "user", vformat(R"(Protocol "%s://" is built-in.)", protocol));
```
I'd use R strings over escaping here, for all